03_codemanagement/git_learnr/Intro to Git.Rmd

```diff
@@ -0,0 +1,612 @@
+* Set `base` to master and `compare` to your new branch
+* Create the merge request
+* Explore this page: review the included commits and differences in code within the commit hash
+* Complete the merge (may take a series of clicks)
+
+At this point, your master branch will reflect the changes you made in your branch. GitHub may automatically delete your branch. If it doesn't, select the option to delete the branch in GitHub.
+
+E. Now return to the Terminal one last time. 
+
+* `git status` Check if you are still on your branch 
+* `git checkout master` Check out the master branch
+* `git pull origin master` Refreshes your local copy of master with the updated, merged version in GitHub
+* `git branch -d [branchname]` Remove your old branch
+  
+
+
+### Practice Git with Others
+
+With a partner, pick a project you will collaborate on. Repeat Steps A-C of the exercise above so that you have 2 branches on the same project. 
+
+In GitHub, create a pull request between your 2 branches and review the code differences and potential conflicts. Complete the merge request. 
+
+
+## Git Resources
+
+### Git Weekly Manual 
+
+```{r, out.width = "800px"}
+knitr::include_graphics("images/git_full_steps.png")
+```
+
+
+0. Make sure you are in your project repository, either of these two ways: 
+    *  Navigate to your project repo by opening the .Rproj file in Files 
+    *  Use Terminal command line to:
+        -  To check your working directory: `pwd`
+        -  To see contents of folder: `ls -a`
+        -  To move into folder below: `cd foldername`
+        -  To move into folder above: `cd ..`
+1. Make sure your master branch is up to date
+    A.  Go to the master branch: `git checkout master`
+    B.  Update your master branch: `git pull origin master`
+2. Create a new branch (give it a useful name): `git checkout -b my_awesome_task`
+    A.  Give your branch a sensible name about what you are working on
+    B.  Add your name or initials to branch name
+3. Go code and write commits!
+    A.  Stage your code `git add <my file>`
+    B.  Commit your code `git commit -m 'look at all this cool stuff’`
+    C.  Push your branch `git push origin my_awesome_task`
+4. Issue a pull request
+    A.  Have someone (maintainers) review your code
+    B.  (Maintainers) merge the pull request
+        -  You can also checkoff a box that will also delete the branch on the remote
+        -  Or delete the branch manually under the branches view
+    C.  Go back to master: `git checkout master`
+    D.  Pull down your merged code: `git pull origin master`
+    E.  Delete your branch: `git branch -d my_awesome_task` (note it is a lower case d) 
+    F.  Clean up your branches: `git fetch --prune`
+
+### Additional Learning Resources
+
+
+I highly recommend the following resources for additional learning and practice.
+
+* Datacamp https://www.datacamp.com/courses/introduction-to-git-for-data-science
+* GitHub Classroom https://classroom.github.com/
```

Comment:
If not mistaken the GitHub Classroom is for teaching classes using GitHub not actual Git training.
I would include rather:
- https://education.github.com/students/experts (Training in Git for students)